planet/src/main/webapp/js/authentication.js

```diff
@@ -194,3 +234,18 @@ function loadElement(href, elementId) {
     xmlhttp.send();
     element.innerHTML = xmlhttp.responseText;
 }
+
+function openProfile() {
+    if (currentUser === null) {
+        return;
+    }
+    database.ref('users/' + currentUser.uid).once('value').then(function(snapshot) {
+        const params = new URLSearchParams();
+        params.append('email', snapshot.val().email);
+        params.append('name', snapshot.val().name);
+        params.append('phone', snapshot.val().phoneNumber);
+        fetch('user', {method: 'POST', body: params}).then(() => {
+            window.location.href = '/user';
+        });
```

Comment:
Is there a way to not go through the servlet at all? I was thinking we can just direct users to the /user page and get the firebase data in its own js file

Reply:
We can do that, but the data in the profile page change with the current user. We'll need a template to be able to use dynamic HTML or use a Java servlet with a JSP file.
Using a java servlet also complicates things because if I try accessing the page "/user" without going through "places.html", the user data in the servlet will just be "null".